### ISSUE TITLE
Some menu items which display icons are inaccessible with NVDA

### ISSUE THREAD

**Issue by nvdaTrac** [nvaccess/test_nvdaTracImport#231]
**Reported by nithal on 3 Aug 2008 05:57 UTC**
**Blocked by #643**
now i'll try to make the stips which i've used to found the proplem ...   1-  choose any txt file or doc file.  2-  bress the app key or the right click .   3-   go down with airo   keys  ...4-  when the screenreader says "open with" use the right caero then move up and down .  you'll not here the names of the items accipt one item that's in the end of the list which called :  "choose the program..."  or something like this.

many of the screenreaders on the world has this proplem but this proplem isn't found with jaws or hal

**Comment by nvdaTrac** [comment]
**Comment by jteh on 3 Aug 2008 06:35 UTC**
This issue occurs because some menu items which display an icon don't seem to expose their name to MSAA. As far as we are aware, the only way to work around this is to obtain the name using display information. (We believe this is how other commercial screen readers solve this problem.) This requires display hooks. Although we have a very early implementation of display hooks, it is not yet ready for inclusion in NVDA. See #40 for details.

Note that the "Open With" menu in Vista does not seem to suffer from this issue.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 14 Sep 2009 23:05 UTC**
MSDN article that might be useful for application developers wishing to fix this issue in their applications: [Exposing Owner-Drawn Menu Items](http://msdn.microsoft.com/en-us/library/dd318076%28VS.85%29.aspx)

**Comment by nvdaTrac** [comment]
**Comment by mdcurran on 10 May 2010 07:15 UTC**
All should be solved by using displayModel.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 13 May 2010 04:05 UTC**
Implemented via display hooking merged in c90061b3e7e8c44aaefae4d29b5839e715ba8ddd.

**Comment by nvdaTrac** [comment]
**Modified by jteh on 13 May 2010 04:07 UTC**
